bpdm-gate/src/main/kotlin/org/eclipse/tractusx/bpdm/gate/service/BusinessPartnerService.kt

```diff
@@ -272,4 +280,51 @@ class BusinessPartnerService(
 
     }
 
+    @Scheduled(cron = "\${cleaningService.pollingCron:-}", zone = "UTC")
+    fun checkBpnUpdates() {
+
+        val poolPaginationRequest = PaginationRequest(0, 100)
+
+        val syncRecord = syncRecordService.getOrCreateRecord(SyncType.POOL_TO_GATE_OUTPUT)
+
+        val changelogSearchRequest = ChangelogSearchRequest(syncRecord.finishedAt)
+
+
+        val poolChangelogEntries = poolClient.changelogs.getChangelogEntries(changelogSearchRequest, poolPaginationRequest)
+        val poolUpdatedEntries = poolChangelogEntries.content.filter { it.changelogType.name == PoolChangeLogType.UPDATE.name }
+
+
+        val bpnA =
+            poolUpdatedEntries.filter { it.businessPartnerType == BusinessPartnerType.ADDRESS && it.changelogType == PoolChangeLogType.UPDATE }.map { it.bpn }
+        val bpnL = poolUpdatedEntries.filter { it.businessPartnerType == BusinessPartnerType.LEGAL_ENTITY && it.changelogType == PoolChangeLogType.UPDATE }
+            .map { it.bpn }
+        val bpnS =
+            poolUpdatedEntries.filter { it.businessPartnerType == BusinessPartnerType.SITE && it.changelogType == PoolChangeLogType.UPDATE }.map { it.bpn }
+
+        val gateOutputEntries = businessPartnerRepository.findByStageAndBpnLInAndBpnSInAndBpnAIn(StageType.Output, bpnL, bpnS, bpnA)
+
+        val businessPartnerGenericDtoList = gateOutputEntries.map { bp ->
+            orchestratorMappings.toBusinessPartnerGenericDto(bp)
+        }
+
+        val taskIds = createGoldenRecordTasks(TaskMode.UpdateFromPool, businessPartnerGenericDtoList).createdTasks.map { it.taskId }
+
+        val pendingRequests = gateOutputEntries.zip(taskIds)
+            .map { (partner, taskId) ->
+                SharingStateService.PendingRequest(
+                    SharingStateService.SharingStateIdentifierDto(
+                        partner.externalId,
+                        partner.parentType!!
+                    ), taskId
+                )
+            }
+        sharingStateService.setPending(pendingRequests)
+
+        if (poolUpdatedEntries.isNotEmpty()) {
+            syncRecord.finishedAt = poolUpdatedEntries.last().timestamp
+            syncRecord.status = SyncStatus.SUCCESS
+            syncRecordService.setSyncRecord(syncRecord)
```

Comment:
Ok, I assume you don't want to use the setSuccess method here because you want to overwrite the finishedAt timestamp. I think it would be cleaner and less error prone if you make the finishedAt-Overwrite an optional parameter in the setSuccess method and use that method here